internal/report/source.go

```diff
@@ -917,8 +918,19 @@ func openSourceFile(path, searchPath, trim string) (*os.File, error) {
 		f, err := os.Open(path)
 		return f, err
 	}
+	possibleBases := filepath.SplitList(searchPath)
+	if gopath := os.Getenv("GOPATH"); gopath != "" {
+		// We can also look through:
+		// * $GOPATH/pkg/mod
+		// * runtime.GOROOT()/src
+		// in case the file originates from Go modules.
+		// See https://github.com/google/pprof/issues/611.
+		possibleBases = append(possibleBases,
+			filepath.Join(gopath, "pkg", "mod"),
+			filepath.Join(runtime.GOROOT(), "src"))
+	}
```

Comment:
Perhaps nitpicking slightly, but this is not entirely correct; the only way to reliably find these out would be to run `go env -json GOMODCACHE GOROOT`.

Using just `os.Getenv("GOPATH")` will not see if the user set up their own `GOMODCACHE`, nor will it work with env vars set with `go env -w`.